enhancements/external-secrets-operator/external-secrets-operator.md

```diff
@@ -0,0 +1,549 @@
+external-secrets is responsible for syncing secrets from external APIs to Kubernetes secrets and managing their lifecycle. The applications 
+dependent on the secrets need not worry about authentication/access, or fetching the secrets from external sources. This enhancement describes
+how the external-secrets project can be deployed by an optional operator on OpenShift.
+
+Note: Throughout the document, the following terminology means.
+- `external-secrets-operator` is the dedicated operator in Red Hat OpenShift for managing the external-secrets project deployment.
+- `external-secrets` is the operand managed by the external-secrets-operator.
+- `externalsecretsoperator.operator.openshift.io` and `externalsecrets.operator.openshift.io` are the custom resources for interacting 
+  with `external-secrets-operator` to install, configure and uninstall the `external-secrets` deployment in Red Hat OpenShift.
+
+## Motivation
+
+Customers often rely on external secret managers (e.g., AWS Secrets Manager, HashiCorp Vault, GCP Secret Manager) to meet security, 
+compliance, and operational standards. The External Secrets Operator (ESO) integrates these systems with Kubernetes in a secure and 
+automated way. ESO acts as a controller that authenticates with external secret stores, retrieves secrets, and injects them into Kubernetes 
+as native Secret resources — without requiring applications to directly access or manage those secrets, which ensures
+- Decouple applications from secret lifecycle management.
+- Ensure compliance by centralizing secret storage.
+- Enable secure, automated secret rotation.
+- Support multi-cloud secret sourcing with fine-grained access control.
+- Access control is centralized and auditable.
+
+### User Stories
+
+- As a cluster administrator, I want the applications to make use of the local references of external secrets, stored as kubernetes secrets 
+  to reduce the overhead of each application accessing the external secret managers.
+- As a cluster administrator, I want to restrict the application’s access to only required secrets stored in the external secret managers, 
+  by making those available as kubernetes secrets.
+- As an application developer, I want my application to not be aware of the external secrets manager used for storing the secrets and also 
+  the authentication mechanism needed to access each external provider.
+
+### Goals
+
+- Allow cluster administrators to install, upgrade and uninstall external-secrets though an optional OLM day-2 operator.
+- New custom resource(CR) `externalsecrets.operator.openshift.io` to be made available to manage external-secrets deployment.
+
+### Non-Goals
+
+- This enhancement will not diverge the `external-secrets` from upstream code, any missing functionality must go through the upstream process.
+- Removing `externalsecrets.operator.openshift.io` CR object will not remove external-secrets deployment. But will only stop the reconciliation of 
```

Comment:
Add upgrade to non-goals, unless it is planned to support.

Reply:
suggestion incorporated.